prow/kube/prowjob.go

```diff
@@ -68,11 +70,11 @@ const (
 )
 
 type ProwJob struct {
-	APIVersion string        `json:"apiVersion,omitempty"`
-	Kind       string        `json:"kind,omitempty"`
-	Metadata   ObjectMeta    `json:"metadata,omitempty"`
-	Spec       ProwJobSpec   `json:"spec,omitempty"`
-	Status     ProwJobStatus `json:"status,omitempty"`
+	metav1.TypeMeta   `json:",inline"`
```

Comment:
Hrm, I just realized this is not supported in Go yet: https://github.com/golang/go/issues/6213

k8s is using a forked version of the json package.

Reply:
Seems that k8s is using https://github.com/json-iterator/go for JSON marshalling/unmarshalling.

Reply:
@deads2k @sttts what JSON library am I supposed to use outside of k8s in order to inline typemeta?

Reply:
Or this is something I shouldn't care about since it's supposedly handled by client-go?

Reply:
> @deads2k @sttts what JSON library am I supposed to use outside of k8s in order to inline typemeta?

jsoniter doesn't handle some cases of embedding.  As I recall, it didn't work for openshift downstream.  In theory, I think client-go will sort it for you.  If not, I'd ask @thockin